scripts/populate_tox/populate_tox.py

```diff
@@ -0,0 +1,502 @@
+    custom_supported_versions: Optional[SpecifierSet] = None,
+) -> list[Version]:
+    """Get an intersection of python_versions and Python versions supported in the SDK."""
+    supported = []
+
+    curr = MIN_PYTHON_VERSION
+    while curr <= MAX_PYTHON_VERSION:
+        if curr in package_python_versions:
+            if not custom_supported_versions or curr in custom_supported_versions:
+                supported.append(curr)
+
+        next = [int(v) for v in str(curr).split(".")]
+        next[1] += 1
+        curr = Version(".".join(map(str, next)))
+
+    return supported
+
+
+def pick_python_versions_to_test(python_versions: list[Version]) -> list[Version]:
+    filtered_python_versions = {
+        python_versions[0],
+    }
+
+    filtered_python_versions.add(python_versions[-1])
+    try:
+        filtered_python_versions.add(python_versions[-2])
+    except IndexError:
+        pass
+
+    return sorted(filtered_python_versions)
+
+
+def determine_python_versions(pypi_data: dict) -> Union[SpecifierSet, list[Version]]:
+    try:
+        classifiers = pypi_data["info"]["classifiers"]
+    except (AttributeError, KeyError):
+        # This function assumes `pypi_data` contains classifiers. This is the case
+        # for the most recent release in the /{project} endpoint or for any release
+        # fetched via the /{project}/{version} endpoint.
+        return []
+
+    python_versions = []
+    for classifier in classifiers:
+        if classifier.startswith(CLASSIFIER_PREFIX):
+            python_version = classifier[len(CLASSIFIER_PREFIX) :]
+            if "." in python_version:
+                # We don't care about stuff like
+                # Programming Language :: Python :: 3 :: Only,
+                # Programming Language :: Python :: 3,
+                # etc., we're only interested in specific versions, like 3.13
+                python_versions.append(Version(python_version))
+
+    if python_versions:
+        python_versions.sort()
+        return python_versions
+
+    # We only use `requires_python` if there are no classifiers. This is because
+    # `requires_python` doesn't tell us anything about the upper bound, which
+    # depends on when the release first came out
+    try:
+        requires_python = pypi_data["info"]["requires_python"]
+    except (AttributeError, KeyError):
+        pass
+
+    if requires_python:
+        return SpecifierSet(requires_python)
+
+    return []
+
+
+def _render_python_versions(python_versions: list[Version]) -> str:
+    return (
+        "{"
+        + ",".join(f"py{version.major}.{version.minor}" for version in python_versions)
+        + "}"
+    )
+
+
+def _render_dependencies(integration: str, releases: list[Version]) -> list[str]:
+    rendered = []
+
+    if TEST_SUITE_CONFIG[integration].get("deps") is None:
+        return rendered
+
+    for constraint, deps in TEST_SUITE_CONFIG[integration]["deps"].items():
```

Comment:
We should not need the `if` statement, as long as we specify a default value of an empty dictionary when getting `deps`

```suggestion
    for constraint, deps in TEST_SUITE_CONFIG[integration].get("deps", {}).items():
```

Reply:
This is really just down to personal preference. I prefer the explicit `if` check because it captures the logical flow of the function better (i.e., if we don't have `deps`, there is nothing to do)